File: 1.Arrays/Solution/Python/Leetcode/128.LongestConsecutiveSequence.py
def longestConsecutive(nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    hashmap = dict()
    result = 0
    for n in nums:
        if (n not in hashmap):
            left = hashmap[n - 1]  if n - 1 in hashmap else 0
            right = hashmap[n + 1] if n + 1 in hashmap else 0
            temp = left + right + 1
            result = max(result, temp)
            hashmap[n] = temp
            hashmap[n - left] = temp
            hashmap[n + right] = temp

    return result

File: 01.Arrays/Leetcode/C++/128.LongestConsecutiveSequence.cpp
#include <algorithm>
#include <unordered_map>
#include <vector>

using namespace std;

int longestConsecutive(const vector<int>& nums) {
    std::unordered_map<int, int> map;
    int alen = nums.size();
    int result = 0;
    for (int i = 0; i < alen; i++) {
        if (map.find(nums[i]) == map.end()) {
            int cur   = nums[i],
                left  = (map.find(cur - 1) != map.end())  ? map[cur - 1] : 0,
                right = (map.find(cur + 1) != map.end()) ? map[cur + 1] : 0;

            int sum = left + right + 1;
            map[cur] = sum;
            result = max(sum, result);

            map[cur - left] = sum;
            map[cur + right] = sum;
        }
    }

    return result;
}
